idaes/core/util/tests/test_model_diagnostics.py

```diff
@@ -185,6 +185,129 @@ def test_vars_near_zero(model):
+
+    @pytest.fixture(scope="function")
+    def model(self):
+        m = ConcreteModel()
+        m.u = Var()
+        m.w = Var(initialize=1)
+        m.x = Var(initialize=1)
+        m.y = Var(range(3), initialize=[1, None, 2])
+        m.z = Var()
+
+        m.con_w = Constraint(expr=m.w == 0)
+        m.con_x = Constraint(expr=m.x == 1)
+
+        def rule_con_y(b, i):
+            return b.y[i] == 3
+
+        m.con_y = Constraint(range(3), rule=rule_con_y)
+        m.con_z = Constraint(expr=m.x + m.z == -1)
+
+        m.block1 = Block()
+        m.block1.a = Var()
+        m.block1.b = Var(initialize=7)
+        m.block1.c = Var(initialize=-5)
+        m.block1.d = Var()
+
+        m.block1.con_a = Constraint(expr=m.block1.a**2 + m.block1.b == 1)
+        m.block1.con_b = Constraint(expr=m.block1.b + m.block1.c == 3)
+        m.block1.con_c = Constraint(expr=m.block1.c + m.block1.a == -4)
+        return m
+
+    @pytest.mark.unit
+    def test_verify_active_variables_initialized(self, model):
+        diag_tbx = DiagnosticsToolbox(model)
+        with pytest.raises(
+            RuntimeError,
+            match=self.err_msg(3),
+        ):
+            diag_tbx._verify_active_variables_initialized()
+
+        model.block1.deactivate()
+        with pytest.raises(
+            RuntimeError,
+            match=self.err_msg(2),
+        ):
+            diag_tbx._verify_active_variables_initialized()
+
+        model.con_y[1].deactivate()
+        model.con_z.deactivate()
+        # No uninitialized variables in remaining constraints
+        diag_tbx._verify_active_variables_initialized()
+
+    @pytest.mark.unit
+    def test_error_handling(self, model):
+        diag_tbx = DiagnosticsToolbox(model)
+        methods = [
+            "display_variables_with_extreme_jacobians",
+            "display_constraints_with_extreme_jacobians",
+            "display_extreme_jacobian_entries",
+            "display_near_parallel_constraints",
+            "display_near_parallel_variables",
+            "display_constraints_with_mismatched_terms",
+            "display_constraints_with_canceling_terms",
+            "display_constraints_with_no_free_variables",
+            "report_numerical_issues",
+        ]
+        for mthd in methods:
+            mthd_obj = getattr(diag_tbx, mthd)
+            with pytest.raises(
+                RuntimeError,
+                match=self.err_msg(3),
+            ):
+                mthd_obj()
+
+    @pytest.mark.unit
+    def test_display_problematic_constraint_terms(self, model):
+        """
+        This method needs to be split in a separate function
+        because it takes a constraint as an argument.
+        """
+        diag_tbx = DiagnosticsToolbox(model)
+        with pytest.raises(
+            RuntimeError,
+            match=self.err_msg(3),
+        ):
+            diag_tbx.display_problematic_constraint_terms(model.con_y[1])
+        # Right now the error message should show whether or not
```

Comment:
Does this mean it will be changed in the future?

Reply:
I don't know.